server/artifacts/artifact_server.go

```diff
@@ -186,41 +182,28 @@ func (a *ArtifactServer) returnArtifact(ctx context.Context, w http.ResponseWrit
 	if err != nil {
 		return err
 	}
-	tmp, err := ioutil.TempFile("/tmp", "artifact")
-	if err != nil {
-		return err
-	}
-	tmpPath := tmp.Name()
-	defer func() { _ = os.Remove(tmpPath) }()
-
-	err = driver.Load(art, tmpPath)
-	if err != nil {
-		return err
-	}
 
-	file, err := os.Open(filepath.Clean(tmpPath))
+	stream, err := driver.OpenStream(art)
 	if err != nil {
 		return err
 	}
 
 	defer func() {
-		if err := file.Close(); err != nil {
-			log.Fatalf("Error closing file[%s]: %v", tmpPath, err)
+		if err := stream.Close(); err != nil {
+			log.Fatalf("Error closing stream[%s]: %v", stream, err)
 		}
 	}()
 
-	stats, err := file.Stat()
-	if err != nil {
-		return err
-	}
-
-	contentLength := strconv.FormatInt(stats.Size(), 10)
-	log.WithFields(log.Fields{"size": contentLength}).Debug("Artifact file size")
-
 	key, _ := art.GetKey()
 	w.Header().Add("Content-Disposition", fmt.Sprintf(`filename="%s"`, path.Base(key)))
 
-	http.ServeContent(w, r, "", time.Time{}, file)
+	_, err = io.Copy(w, stream)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
I don't think this line is needed? the calling function should capture the error and do it

Reply:
Curious how the calling function knows which type of status based on error (i.e. 400 level vs 500 level), but I can take a look

Reply:
I see the caller is assuming it's a 500 error. That's fine in this case at least so I'll use it.

Reply:
I presume the call to writeHeader(http.StatusOK) below is probably also unnecessary...

Reply:
check the call-stack?